cdap-formats/src/main/java/io/cdap/cdap/format/io/StructuredRecordDatumReader.java

```diff
@@ -50,6 +52,17 @@ public StructuredRecord read(Decoder decoder, Schema sourceSchema) throws IOExce
    * @throws IOException If failed to decode
    */
   protected final Object decode(Decoder decoder, Schema schema) throws IOException {
+    //handle any logical types that need special handling
+    if (schema.getLogicalType() != null) {
```

Comment:
validation of the datetime value should happen when building the StructuredRecord (https://cdap.atlassian.net/browse/CDAP-15317) rather than here, as many other bad things can happen if the record holds a bad value. Don't need to do it in this PR, as it's covered by that other jira.

Encoder/decoder should be able to just look at the physical type.

Reply:
Removed .